src/autoreplace_cmd.cpp

```diff
@@ -470,6 +471,7 @@ static CommandCost ReplaceFreeUnit(Vehicle **single_unit, DoCommandFlag flags, b
 			*single_unit = new_v;
 
 			AI::NewEvent(old_v->owner, new ScriptEventVehicleAutoReplaced(old_v->index, new_v->index));
+			Game::NewEvent(new ScriptEventVehicleAutoReplaced(old_v->index, new_v->index));
```

Comment:
I'd almost template creating a new event in a manner that creates two events to keep the practical code duplication relatively low.

That would be something like:
```
class Script {
public:
    template <class EventClass, typename ... Args> 
    static void NewEvent(int owner, Args... args) {
        AI::NewEvent(owner, new EventClass(args...));
        Game::NewEvent(new EventClass(args...));
    }
};
```
and then `Script::NewEvent<ScriptEventVehicleAutoReplaced>(old_v->owner, old_v->index, new_v->index);`. Yes, when a company needs to be passed to the constructor you pass the owner twice in a row, or you make a second templated function `Script::NewEventWithOwner` or something that under water passes the owner twice to `Script::NewEvent`.

I'm not sure whether either way with the templating is the best solution though.